### PR TITLE
Export ImageKit interfaces

### PR DESCRIPTION
**Pull request by mattleff**
This PR adds support for importing the ImageKit interfaces into other TS code.

----

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Comment by mattleff** [comment]
@sandersn The CI failure is a test-all failure unrelated to my changes. Thanks for taking a look!

**Comment by jablko** [comment]
@mattleff The CI failure fell in the window between releasing [@definitelytyped/definitions-parser v0.0.54](https://github.com/microsoft/DefinitelyTyped-tools/commit/dfb84eb1d27d9db46e92fcd2490ab41525104db7), which includes microsoft/DefinitelyTyped-tools#93, and landing #44091, which I've just marked ready for review.

The DefinitelyTyped-tools change enables it to find files that previously required exceptions, and #44091 removes those exceptions. The changes are interdependent: Regardless of which lands first, there's a window of CI failure. Apologies!

**Comment by sandersn** [comment]
master is in a good state now, but it looks like you'll have to merge from master to get CI passing. Sorry about the break.

**Review by romdim — APPROVED** [review]

File: types/imagekit/tslint.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-    "extends": "dtslint/dt.json"
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-redundant-jsdoc-2": false
```
Comment (sandersn) [inline types/imagekit/tslint.json:4]: This rule crashes on `@see` right now, but shouldn't fail CI. Can you revert this for now?

(I merged a fix this morning but haven't shipped it yet.)

**Review by sandersn — CHANGES_REQUESTED** [review]

**Review by romdim — APPROVED** [review]

**Comment by mattleff** [comment]
Ready to merge

File: types/imagekit/index.d.ts [hunk 2]
```diff
@@ -6,929 +6,948 @@
...
+
+        /**
+         * This API can list all the uploaded files in your ImageKit.io media library.
+         * For searching and filtering, you can use query parameters as described below.
+         *
+         * @see {@link https://docs.imagekit.io/api-reference/media-api/list-and-search-files}
+         *
+         * @param listFilesOptions
+         */
+        listFiles(listFilesOptions: ImageKit.ListFileOptions): Promise<ImageKit.ListFileResponse>;
```
Comment (gregegan) [inline types/imagekit/index.d.ts:754]: It looks like this should be `Promise<ImageKit.ListFileResponse[]>`

https://docs.imagekit.io/api-reference/media-api/list-and-search-files

`The JSON-encoded response has an array of items. Each item can have the following properties.`

**Pull request merged**